src/mesh/rbf.f90

```diff
@@ -0,0 +1,597 @@
+           + Elem%CurvedNode(HexaMapInv(N,N,N))%np%x
+  xBary(:) = xBary(:) / 8.
+  ! Cycle if the bary center is not inside of the bounding box (thus only all or no points of an element are curved)
+  IF ((xBary(1).LT.xMin).OR.(xBary(1).GT.xMax).OR.(xBary(2).LT.yMin).OR.(xBary(2).GT.yMax)) THEN
+    Elem=>Elem%nextElem
+    CYCLE
+  END IF
+  ! Then, apply Vandermonde matrix, e.g. evaluate bi-linear mapping on all curved side nodes
+  xTriLinear(1,:) = MATMUL(Vdm_VolTriLinear,xCornerVol(1,:))
+  xTriLinear(2,:) = MATMUL(Vdm_VolTriLinear,xCornerVol(2,:))
+  xTriLinear(3,:) = MATMUL(Vdm_VolTriLinear,xCornerVol(3,:))
+  DO i=1,Elem%nCurvedNodes
+    ! Evaluate the RBF interpolation at each linear volume point
+    x = xTriLinear(:,i)
+    xTmp = x
+    DO iBP=1,nBP
+      dist = Distance(x,RefCoordinates(:,iBP))
+      RBFValue = EvaluateRBF(dist,supportRadius(iRBFBox),RBFType(iRBFBox))
+      xTmp(1) = xTmp(1) + RBFValue*RBFRHS(iBP,1)
+      xTmp(2) = xTmp(2) + RBFValue*RBFRHS(iBP,2)
+      xTmp(3) = xTmp(3) + RBFValue*RBFRHS(iBP,3)
+    END DO
+    ! Take linear polynomial into account
+    xTmp = xTmp + RBFRHS(nBP+1,:) ! constant part
+    xTmp(1) = xTmp(1) + DOT_PRODUCT(RBFRHS(nBP+2:nBP+4,1),x(:)) ! Linear part
+    xTmp(2) = xTmp(2) + DOT_PRODUCT(RBFRHS(nBP+2:nBP+4,2),x(:)) ! Linear part
+    xTmp(3) = xTmp(3) + DOT_PRODUCT(RBFRHS(nBP+2:nBP+4,3),x(:)) ! Linear part
+    ! Overwrite curved node position by interpolation
+    Elem%CurvedNode(i)%np%x = xTmp
+  END DO
+  Elem=>Elem%nextElem
+END DO
+
+DEALLOCATE(RBFRHS)
+WRITE(UNIT_stdOut,'(A)')' VOLUME CURVING DONE!'
+
+END SUBROUTINE RBFVolumeCurving
+
+
+
+FUNCTION EvaluateRBF(dist,supportRadius,RBFType)
+!===================================================================================================================================
+! Evaluates the choosen RBF at distance dist with support radius supportRadius
+! Numbering and choice of RBFs is taken from: A. de Boer et al. / Computers and Structures 85 (2007) 784–795
+! The multiquadratic biharmonic RBFs (type 10 and 11) take an additional parameter (a) which controls the shape of the function.
+! Typical values for this parameter are in the range 10E-5 - 10E-3.
+!===================================================================================================================================
+USE MOD_Globals
+! INPUT VARIABLES
+REAL,INTENT(IN)    :: dist          ! Euclidean distance used in RBF evaluation
+REAL,INTENT(IN)    :: supportRadius ! Normalizing constant for RBF evaluation
+INTEGER,INTENT(IN) :: RBFType       ! Type of RBF to evaluate
+!-----------------------------------------------------------------------------------------------------------------------------------
+! OUTPUT VARIABLES
+REAL            :: EvaluateRBF
+!-----------------------------------------------------------------------------------------------------------------------------------
+! LOCAL VARIABLES
+REAL             :: xi
+REAL,PARAMETER   :: a=10E-3
```

Comment:
Is there a justification for the value chosen here?

Reply:
Yes, in the comments above. Seems to be shape parameter. Added comment to this line to make it very clear.